### PR TITLE
JWT validation in forum sample

### PR DESCRIPTION
**Pull request by letmaik**
This also removes the guest user mode on the website. For tests and bulk import during demos a separate JWT signing key is generated and imported to avoid having to use real Microsoft identities.

### CODE REVIEW

File: samples/apps/forum/src/authentication.ts [hunk 1]
```diff
@@ -21,12 +42,74 @@ export function authentication(request: ccf.Request, securityName: string, scope
             throw new UnauthorizedError('unexpected authentication type')
```
Comment (jumaffre) [inline samples/apps/forum/src/authentication.ts:42]:
```suggestion
            throw new UnauthorizedError('unexpected authentication type ${parts[0]}, expected Bearer')
```

File: samples/apps/forum/src/controllers/site.ts [hunk 2]
```diff
@@ -41,20 +41,27 @@ const HEADER_HTML = `
...
 const siteUrl = window.location.origin + '/app/site'
 
+// "CCF Demo App" app registration
+const MS_APP_ID = "1773214f-72b8-48f9-ae18-81e30fab04db"
+// Note: Even though a scope is not strictly required for this app,
+// it is needed so that Microsoft produces standard-compliant JWTs.
+// Creating and requesting a scope of our own app changes the token audience
+// to our app instead of being Microsoft (which is the default).
+// See also https://xsreality.medium.com/making-azure-ad-oidc-compliant-5734b70c43ff.
+const MS_LOGIN_SCOPES = ["api://1773214f-72b8-48f9-ae18-81e30fab04db/Polls.Access"]
```
Comment (jumaffre) [inline samples/apps/forum/src/controllers/site.ts:60]: Perhaps add a link to https://docs.microsoft.com/en-us/graph/auth-register-app-v2, which is a pre-requisite to get any similar sample app working?

**Review by jumaffre — APPROVED** [review]

File: samples/apps/forum/src/controllers/site.ts (hunk 2 above)
Comment (letmaik) [reply samples/apps/forum/src/controllers/site.ts:61]: This is for Graph API specifically which may have confusing bits. I added a link to https://docs.microsoft.com/en-us/azure/active-directory/develop/scenario-protected-web-api-overview instead which is likely a better landing page.

File: samples/apps/forum/src/authentication.ts [hunk 3]
```diff
@@ -10,23 +12,104 @@ export interface User {
...
+    iss: string
+}
+
+interface MSAccessTokenClaims extends BodyClaims {
+    aud: string
+    appid: string // 1.0 only
+    ver: string // 1.0 or 2.0
+}
+
+const MS_APP_ID = "1773214f-72b8-48f9-ae18-81e30fab04db"
```
Comment (achamayou) [inline samples/apps/forum/src/authentication.ts:30]: A comment with a link explaining how to get your own MS_APP_ID may be worthwhile here. I suggest https://docs.microsoft.com/en-us/powerapps/developer/common-data-service/walkthrough-register-app-azure-active-directory

**Review by achamayou — APPROVED** [review]

**Comment by letmaik** [comment]
@achamayou Can you fix the merge conflicts here? The idea was to merge this PR before the prettier PR :)

**Comment by achamayou** [comment]
@letmaik sorry, I was not aware this was the idea. For what it's worth, `./scripts/ci-checks.sh -f` will automatically format everything happily.

**Comment by letmaik** [comment]
@achamayou All green :)

**Pull request merged**